datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -260,7 +258,11 @@ impl<F: FileOpener> FileStream<F> {
                         self.file_stream_metrics.time_scanning_until_data.stop();
                         self.file_stream_metrics.time_scanning_total.stop();
                         let result = result
-                            .and_then(|b| self.pc_projector.project(b, partition_values))
+                            .and_then(|b| {
+                                self.pc_projector
+                                    .project(b, partition_values)
+                                    .map_err(|e| ArrowError::ExternalError(e.into()))
```

Comment:
Is this needed, `project` already returns `ArrowError` AFAICT?

Reply:
This is project from DataFusion, it returns DataFusionError